### PR TITLE
Do not report an error and eat `scoped` identifier following `ref` keyword.

### LINKED ISSUE #63986: Breaking change around contextual keyword `scoped`
The following code used to compile without errors.
```
class Test
{
    static void Main()
    {
        scoped a;
        ref scoped c = ref a;
    }
}

struct @scoped
{}
```

Observed:
```
error CS9061: Unexpected contextual keyword 'scoped'. Did you mean 'scoped ref' or '@scoped'?
error CS1001: Identifier expected
error CS0246: The type or namespace name 'c' could not be found (are you missing a using directive or an assembly reference?)
error CS8173: The expression must be of type 'c' because it is being assigned by reference
```

Note that ```scoped a;``` doesn't cause any errors, So, adding ```ref``` in front of it should not cause any errors either.

I think we should either remove CS9061 completely, or report it only in situations when interpreting `scoped` as an identifier would cause an error too.   

Note that there is no similar error for a parameter:
```
class Test
{
    static void M(scoped a, ref scoped c)
    {
    }
}

struct @scoped
{}
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #63986.

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs [hunk 1]
```diff
@@ -2109,7 +2109,7 @@ internal enum ErrorCode
         ERR_FeatureNotAvailableInVersion11 = 9058,
         ERR_RefFieldInNonRefStruct = 9059,
         ERR_CannotMatchOnINumberBase = 9060,
-        ERR_MisplacedScoped = 9061,
+        // Available 9061,
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Errors/ErrorCode.cs:2112]: We shipped this error in 17.4-preview1. Is it fine to reuse it at this point?

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Errors/ErrorCode.cs:2112]: > We shipped this error in 17.4-preview1. Is it fine to reuse it at this point?

I think everything until RTM is a fair game

**Comment by AlekseyTs** [comment]
@cston, @jcouv, @dotnet/roslyn-compiler For the second review.

**Comment by AlekseyTs** [comment]
@cston, @jcouv, @dotnet/roslyn-compiler For the second review.

**Comment by AlekseyTs** [comment]
@cston, @jcouv, @dotnet/roslyn-compiler For the second review.  Majority of changes are in tests.

**Review by cston — APPROVED** [review]

**Pull request merged**